model/Tree.java

```diff
@@ -1,5 +1,257 @@
+	{
+		dependencies = new Properties();
+	
+		myRegistry.setDependencies(dependencies);
+	}
+	
+	//---------------------------------------------------------------------
+	public Object getState(String key)
+	{
+		if (key.equals("UpdateStatusMessage") == true)
+			return updateStatusMessage;
+
+		return persistentState.getProperty(key);
+	}
+
+	//---------------------------------------------------------------------
+	public void stateChangeRequest(String key, Object value)
+	{
+		if(key.equals("ProcessTree"))
+		{
+			processNewTree((Properties) value);
+		}
+
+		myRegistry.updateSubscribers(key, this);
+	}
+
+	//---------------------------------------------------------------------
+	public void updateState(String key, Object value)
+	{
+		stateChangeRequest(key, value);
+	}
+	
+	public void update()
+	{
+		updateStateInDatabase();
+	}
+	
+	//---------------------------------------------------------------------
+	private void updateStateInDatabase() 
+	{
+		try
+		{
+			if (persistentState.getProperty("Barcode") != null)
+			{
+				Properties whereClause = new Properties();
+				whereClause.setProperty("Barcode",
+				persistentState.getProperty("Barcode"));
+				updatePersistentState(mySchema, persistentState, whereClause);
+				updateStatusMessage = "Tree data for barcode : " + persistentState.getProperty("Barcode") + " updated successfully in database!";
+			}
+			else
+			{
+				int bookId =
+					insertAutoIncrementalPersistentState(mySchema, persistentState);
+				persistentState.setProperty("Barcode", "" + Barcode);
+				updateStatusMessage = "Tree data for new tree : " +  persistentState.getProperty("Barcode" + "installed successfully in database!");
+			}
+		}
+		catch (SQLException ex)
+		{
+			updateStatusMessage = "Error in installing tree data in database!";
+		}
+		//DEBUG System.out.println("updateStateInDatabase " + updateStatusMessage);
+	}
+	
+	//---------------------------------------------------------------------
+	public Vector<String> getEntryListView()
+	{
+		Vector<String> v = new Vector<String>();
+
+		v.addElement(persistentState.getProperty("Barcode"));
+		v.addElement(persistentState.getProperty("Type"));
+		v.addElement(persistentState.getProperty("Notes"));
+		v.addElement(persistentState.getProperty("Status"));
+		v.addElement(persistentState.getProperty("DateStatusUpdated"));
+
+		return v;
+	}
+	
+	//---------------------------------------------------------------------
+	protected void initializeSchema(String tableName)
+	{
+		if (mySchema == null)
+		{
+			mySchema = getSchemaInfo(tableName);
+		}
+	}
+
+	public String toString()
+	{
+		return "Barcode: " + persistentState.getProperty("Barcode") +
+				"; Type: " + persistentState.getProperty("Type") +
+				"; Notes: " + persistentState.getProperty("Notes") +
+				"; Status: " + persistentState.getProperty("Status") +
+				"; DateStatusUpdated: " + persistentState.getProperty("DateStatusUpdated") + "\n";
+	}
+	public void createAndShowTreeView()
```

Comment:
views should be handled by the TLC not here, whenever you need to change views you should call the stateChangeRequest method of the TLC and add an entry to that method to handle the request, I can show you in class what I mean if you are confused